### PR TITLE
Do not perform narrow_exprs() on floating point and double precision floats

### PR DESCRIPTION
**Pull request by AOS002**
We should not be narrowing exprs that are floating point and double precision floats

### CODE REVIEW

File: angr/analyses/decompiler/ail_simplifier.py [hunk 1]
```diff
@@ -233,6 +233,9 @@ def _narrow_exprs(self) -> bool:
         sorted_defs = sorted(rd.all_definitions, key=lambda d: d.codeloc, reverse=True)
         for def_ in (d_ for d_ in sorted_defs if d_.codeloc.context is None):
             if isinstance(def_.atom, atoms.Register):
+                # we should not narrow floating point and double precision floats
+                if self.project.arch.translate_register_name(def_.atom.reg_offset, def_.atom.size).startswith("xmm"):
+                    continue
```
Comment (ltfish) [inline angr/analyses/decompiler/ail_simplifier.py:238]: This is not architecture agnostic. Please avoid x86-specific logic :)

**Comment by ltfish** [comment]
Looks like the test cases are failing?

**Comment by AOS002** [comment]
got the test cases to succeed by ignoring the artificial registers from the check

**Comment by ltfish** [comment]
Thank you for the fix!

**Pull request merged**